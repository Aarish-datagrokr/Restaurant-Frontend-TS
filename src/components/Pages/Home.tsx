import './styles/home.css'
import {ReactTypical} from '@deadcoder0904/react-typical';

const Home = () => {
        var status = "Open!";
        var about = "Get your table number by booking a reservation and have a meal at our restaurant. Bookings are closed after 8:00 PM IST.";
        var today = new Date().getHours();
        if(today>=18) { 
            status="Closed!";
            about = "We're sorry but it's past 8:00 PM and bookings are closed, You can book a table for the next day, Booking starts at 12:00 AM IST.";
        }
        return (
        <div style={{marginTop:"80px"}}>
                <ReactTypical
                     steps={['Bookings', 1000, `Bookings ${status}`, 500]}
                     loop={Infinity}
                     wrapper="p"
                    className="welcome-text"
                  />
            <p className="about">{about}</p>
            <p className='slogan'>We make food, you make memories.</p>
        </div>
    );
}

export default Home;