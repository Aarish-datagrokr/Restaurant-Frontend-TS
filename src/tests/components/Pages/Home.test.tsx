import {shallow} from 'enzyme';
import Home from '../../../components/Pages/Home';

test('Should test Home component', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
});
