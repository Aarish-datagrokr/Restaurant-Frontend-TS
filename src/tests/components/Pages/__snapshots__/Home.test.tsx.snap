// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test Home component 1`] = `
<div
  style={
    Object {
      "marginTop": "80px",
    }
  }
>
  <Memo(k)
    className="welcome-text"
    loop={Infinity}
    steps={
      Array [
        "Bookings",
        1000,
        "Bookings Open!",
        500,
      ]
    }
    wrapper="p"
  />
  <p
    className="about"
  >
    Get your table number by booking a reservation and have a meal at our restaurant. Bookings are closed after 8:00 PM IST.
  </p>
  <p
    className="slogan"
  >
    We make food, you make memories.
  </p>
</div>
`;
